src/lines-component.coffee

```diff
@@ -302,29 +322,24 @@ class LinesComponent
 
         continue if char is '\0'
 
-        unless charWidths[char]?
-          unless textNode?
-            rangeForMeasurement ?= document.createRange()
-            iterator =  document.createNodeIterator(lineNode, NodeFilter.SHOW_TEXT, AcceptFilter)
-            textNode = iterator.nextNode()
-            textNodeIndex = 0
-            nextTextNodeIndex = textNode.textContent.length
-
-          while nextTextNodeIndex <= charIndex
-            textNode = iterator.nextNode()
-            textNodeIndex = nextTextNodeIndex
-            nextTextNodeIndex = textNodeIndex + textNode.textContent.length
-
-          i = charIndex - textNodeIndex
-          rangeForMeasurement.setStart(textNode, i)
-          rangeForMeasurement.setEnd(textNode, i + charLength)
-          charWidth = rangeForMeasurement.getBoundingClientRect().width
-          @presenter.setScopedCharacterWidth(scopes, char, charWidth)
-
+        unless textNode?
+          rangeForMeasurement ?= document.createRange()
+          iterator =  document.createNodeIterator(lineNode, NodeFilter.SHOW_TEXT, AcceptFilter)
+          textNode = iterator.nextNode()
+          textNodeIndex = 0
+          nextTextNodeIndex = textNode.textContent.length
+
+        while nextTextNodeIndex <= charIndex
+          textNode = iterator.nextNode()
+          textNodeIndex = nextTextNodeIndex
+          nextTextNodeIndex = textNodeIndex + textNode.textContent.length
+
+        i = charIndex - textNodeIndex
+        rangeForMeasurement.setStart(textNode, i)
+        rangeForMeasurement.setEnd(textNode, i + charLength)
+        charWidth = rangeForMeasurement.getBoundingClientRect().width
```

Comment:
Glad you're using this approach instead of spans. You could leave the start of the measurement range at index 0 of the first text node if you want to store a running total rather than individual widths.